src/transit_equity/networks/network_prep.py

```diff
@@ -1,38 +1,162 @@
+    """
 
-"""
+    # Check database urls are correct
+    print(os.getenv(postgres_url_ng))
 
-import pandas as pd
-import geopandas as gpd
-from transit_equity.geospatial.format_conversions import load_wkb
+    #connect to engines
+    engine_ng = create_engine(os.getenv(postgres_url_ng))
+
+    # Setup Session Maker and Session
+    session_ng_maker = sessionmaker(bind=engine_ng)
+    session_ng = session_ng_maker()
+
+    # NG test Schema Base
+    base_ng_test = get_automap_base_with_views(engine=engine_ng, schema=test_schema)
+    base_ng_orca = get_automap_base_with_views(engine=engine_ng, schema=orca_schema)
+
+    # Tables of interest from orca_ng
+    trips_ng = base_ng_test.metadata.tables[trips_table]
+    alights_ng = base_ng_test.metadata.tables[alights_table]
+    boardings_ng = base_ng_test.metadata.tables[boardings_table]
+    vboardings_ng = base_ng_orca.metadata.tables[vboardings_table]
+    gtfs_stops_ng = base_ng_test.metadata.tables[gtfs_table]
+
+    # Constructing the query
+    query = (
+        session_ng.query(
+            vboardings_ng.c.card_id,
+            vboardings_ng.c.txn_id,
+            alights_ng.c.txn_id,
+            vboardings_ng.c.device_dtm_pacific,
+            alights_ng.c.alight_dtm_pacific,
+            boardings_ng.c.stop_location,
+            gtfs_stops_ng.c.stop_location
+        ).select_from(trips_ng)
+        .join(boardings_ng, boardings_ng.c.txn_id == trips_ng.c.orig_txn_id)
+        .join(alights_ng, alights_ng.c.txn_id == trips_ng.c.dest_txn_id)
+        .join(vboardings_ng, vboardings_ng.c.txn_id == trips_ng.c.orig_txn_id)
+        .join(gtfs_stops_ng, gtfs_stops_ng.c.stop_id == alights_ng.c.stop_id)
+        .filter(
+            and_(
+                boardings_ng.c.stop_location.isnot(None),
+                gtfs_stops_ng.c.stop_location.isnot(None),
+                vboardings_ng.c.passenger_type_id == user_type
+            )
+        )
+    )
+
+    # Because the adults table is so large that it was causing memory limitation issues, read the
+    # table in chunks of 100000 rows and then concatenate them after.
+    chunk_size = 100000
+
+
+    with engine_ng.connect() as connection:
+        result_proxy = connection.execution_options(stream_results=True).execute(query.statement)
+        chunk_count = 0
+        chunks = []
+
+        while True:
+            chunk = result_proxy.fetchmany(chunk_size)
+            if not chunk:
+                break
+            chunk_df = pd.DataFrame(chunk, columns=result_proxy.keys())
+            # Filter and clean each chunk here, can trade out for other cleaning pipeline for other
+            # analyses if desired
+            filtered_chunk_gdf = clean_and_filter_network_data(chunk_df)
+            chunks.append(filtered_chunk_gdf)
+            chunk_count += 1
+            # Print progress for each 10 chunks read.
+            if chunk_count % 10 == 0:
+                print(f"Fetched chunk {chunk_count}")
```

Comment:
This will be necessary. 
In fact, if possible, please do see if you could utilize tqdm for this, so that you could give better feedback to the user on progress.

Reply:
Thanks for this code. 
It will be useful for me too.